### PR TITLE
[TF-397] add documentation for setting python version

### PR DESCRIPTION
**Pull request by realdoug**
This resolves https://bugs.swift.org/browse/TF-397

https://nbviewer.jupyter.org/github/realdoug/tensorflow-swift/blob/TF-397/docs/site/tutorials/python_interoperability.ipynb?invalidatecache=somethingunique

### CODE REVIEW

**Comment by realdoug** [comment]
Just signed the CLA

**Comment by saeta** [comment]
@realdoug this is super great! LGTM.

File: docs/site/tutorials/python_interoperability.ipynb [hunk 1]
```diff
@@ -1,608 +1,667 @@
...
+   "metadata": {},
+   "source": [
+    "## Python Versions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The easiest way to use a particular version of Python is by setting `PYTHON_LIBRARY` in your shell.  For example: \n",
```
Comment (dan-zheng) [inline docs/site/tutorials/python_interoperability.ipynb]:
```suggestion
    "To use a specific Python installation, set the `PYTHON_LIBRARY` environment variable to the `libpython` shared library provided by the installation. For example:\n",
```

File: docs/site/tutorials/python_interoperability.ipynb [hunk 2]
```diff
@@ -1,608 +1,667 @@
...
+    "\n",
+    "`export PYTHON_LIBRARY=\"~/anaconda3/lib/libpython3.7m.so\"`\n",
+    "\n",
+    "If using anaconda, the exact filename will differ on different platforms."
```
Comment (dan-zheng) [inline docs/site/tutorials/python_interoperability.ipynb]: The exact filename will differ across Python environments and platforms.

File: docs/site/tutorials/python_interoperability.ipynb [hunk 3]
```diff
@@ -1,608 +1,667 @@
...
-            "(1, 2, 3)\r\n"
-          ],
-          "name": "stdout"
-        }
-      ]
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Alternatively, you can also set `PYTHON_VERSION`.  This will cause swift to iteratively search the default path for your system for a matching version.  `PYTHON_LIBRARY` holds precedence over `PYTHON_VERSION`.\n",
```
Comment (dan-zheng) [inline docs/site/tutorials/python_interoperability.ipynb]:
```suggestion
    "Alternatively, you can set the `PYTHON_VERSION` environment variable. If specified, Swift searches system library paths for a matching Python version. Note that `PYTHON_LIBRARY` takes precedence over `PYTHON_VERSION`.\n",
```

File: docs/site/tutorials/python_interoperability.ipynb [hunk 4]
```diff
@@ -1,608 +1,667 @@
...
+    "\n",
+    "There is also a `PythonLibrary.useVersion` function which is the equivalent of setting `PYTHON_VERSION`."
```
Comment (dan-zheng) [inline docs/site/tutorials/python_interoperability.ipynb]:
```suggestion
    "In your code, you can also call the `PythonLibrary.useVersion` function, which is equivalent to setting `PYTHON_VERSION`."
```

File: docs/site/tutorials/python_interoperability.ipynb [hunk 5]
```diff
@@ -1,608 +1,667 @@
...
+   "source": [
+    "// PythonLibrary.useVersion(2)\n",
+    "// PythonLibrary.useVersion(3, 7)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "__NOTE: `useVersion` is only effective if run before Python is loaded.__  Python is lazily loaded the first time you invoke a method on the `Python` global."
```
Comment (dan-zheng) [inline docs/site/tutorials/python_interoperability.ipynb]:
```suggestion
    "__Note: you should run `PythonLibrary.useVersion` right after `import Python`, before calling any Python code. It cannot be used to dynamically switch Python versions.__"
```

File: docs/site/tutorials/python_interoperability.ipynb [hunk 6]
```diff
@@ -1,608 +1,667 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Set `PYTHON_LOADER_LOGGING=1` in your shell to see debug output for the [python loader](https://github.com/apple/swift/pull/20674#discussion_r235207008).  "
```
Comment (dan-zheng) [inline docs/site/tutorials/python_interoperability.ipynb]:
```suggestion
    "Set `PYTHON_LOADER_LOGGING=1` to see [debug output for Python library loading](https://github.com/apple/swift/pull/20674#discussion_r235207008).  "
```

**Comment by realdoug** [comment]
@dan-zheng All make sense to me.  I've updated the PR with revisions.

File: docs/site/tutorials/python_interoperability.ipynb [hunk 7]
```diff
@@ -1,608 +1,668 @@
...
+   "metadata": {},
+   "source": [
+    "## Python Versions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "By default, on `import Python`, Swift searches system library paths for the newest version of Python installed. \n",
```
Comment (dan-zheng) [inline docs/site/tutorials/python_interoperability.ipynb]: Sorry, one more minor change:
```suggestion
    "By default, when you `import Python`, Swift searches system library paths for the newest version of Python installed. \n",
```

File: docs/site/tutorials/python_interoperability.ipynb [hunk 8]
```diff
@@ -1,608 +1,668 @@
...
+    "Swift For TensorFlow supports Python interoperability.\n",
+    "\n",
+    "You can import Python modules from Swift, call Python functions, and convert values between Swift and Python."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Python Versions"
```
Comment (dan-zheng) [inline docs/site/tutorials/python_interoperability.ipynb]:
```suggestion
    "## Setting the Python version"
```

File: docs/site/tutorials/python_interoperability.ipynb [hunk 9]
```diff
@@ -1,608 +1,668 @@
...
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "3.6.7 (default, Oct 22 2018, 11:32:17) \r\n",
+      "[GCC 8.2.0]\r\n"
+     ]
     }
+   ],
+   "source": [
```
Comment (dan-zheng) [inline docs/site/tutorials/python_interoperability.ipynb]: Could you please move this code block under the title section, and remove the `// export ...` line? I hope this picture makes things clear:

image

Comment (rxwei) [reply docs/site/tutorials/python_interoperability.ipynb]: +1.

**Comment by realdoug** [comment]
@dan-zheng sounds good.  done & done.

**Pull request merged**